crates/anemo/src/config.rs

```diff
@@ -229,6 +236,10 @@ impl Config {
     pub(crate) fn outbound_request_timeout(&self) -> Option<Duration> {
         self.outbound_request_timeout_ms.map(Duration::from_millis)
     }
+
+    pub(crate) fn shutdown_idle_timeout(&self) -> Option<Duration> {
+        self.shutdown_idle_timeout_ms.map(Duration::from_millis)
+    }
```

Comment:
This shouldn't return a an Option but instead just a Duration, if shutdown_idle_timeout isn't set then it should use the default that you mentioned in the documentation for this parameter.

Reply:
Agreed, I think I wanted to make the `wait_idle` a bit more autonomous in terms of accepting an optional timeout parameter in case some other component would like to use it without defining a timeout, but probably that doesn't make much sense - let me change that